src/dimwits/scale.hpp

```diff
@@ -1,50 +1,57 @@
 namespace scale {
 
-struct Tag{};
+template< typename Prefix, typename BaseKey >
+struct Key : hana::type< hana::basic_tuple< Prefix, BaseKey > >{};
 
-template< typename BaseUnit, typename Identifier >
-struct Template : hana::type< hana::tuple< BaseUnit, Identifier > >{};
+/* the `key` and `type` functions are only used in unevaluated contexts
+ * to deduce a return type, and as such, need not be vetted for performance.
+ */
 
-template< typename Prefix, typename T >
-constexpr auto primitive( Prefix, dimwits::unit::Primitive< T > ){
-  static_assert( std::is_base_of<Tag, Prefix>::value, "" );
-  return Template< dimwits::unit::Primitive< T >, Prefix >{};
+template< typename Prefix, typename BaseKey >
+constexpr auto key( Prefix, BaseKey ){
+  return Key< Prefix, BaseKey >{};
 }
 
+// !!!!!!!!!!!!!!!!!!!!!!!!
+// intentionally undefined
```

Comment:
Why is this intentionally undefined?

Reply:
This overload is the best match if a user tries to add a prefix to a prefix-ed unit, e.g.

```
kilo(mega(gram));
```

Because the overload isn't defined, doing so results in a compilation error.

Will expand the comment to explain